tests/unit/src/main/scala/tests/TestingClient.scala

```diff
@@ -71,6 +71,7 @@ final class TestingClient(workspace: AbsolutePath, buffers: Buffers)
     _: ShowMessageRequestParams =>
       None
   }
+  var inputBoxHandler: MetalsInputBoxParams => Option[MetalsInputBoxResult] = _
```

Comment:
while picking kind of file can be expressed via `showMessageRequest`, asking for name of the new file cannot, so I had to add this handler for testing purposes.

Reply:
Can you add a default implementation returning some random name? You can have `var nameToPick = "Foo"` and replace that if needed. You also have None as the default if it's an issue.

Otherwise, we will get a null pointer in case of some particular changes.